packages/cards-editor-middleware/src/index.ts

```diff
@@ -0,0 +1,130 @@
+                    component: app.id
+                }
+            })
+        );
+    });
+
+    /**
+     * Route to store card manifest files, the files are stored in the webapp folder of the project
+     * and the application manifest.json file is updated with the new card manifests information within the sap.cards.ap.embeds
+     */
+    router.post(ApiRoutes.cardsStore, async (req: Request, res: Response) => {
+        try {
+            const { floorplan, localPath, fileName = 'manifest.json', manifests } = req.body;
+            const webappPath = await getWebappPath(path.resolve());
+            const fullPath = path.resolve(webappPath, localPath);
+
+            if (!fullPath.startsWith(webappPath)) {
+                throw new Error('Invalid path');
+            }
+
+            const file = utils.prepareFileName(fullPath + '/' + fileName);
+            if (!existsSync(fullPath)) {
+                try {
+                    mkdirSync(fullPath, { recursive: true });
+                } catch (err) {
+                    res.status(403).send(`Files could not be created/updated.`);
+                }
+            }
+            const multipleCards = utils.prepareCardTypesForSaving(manifests);
+            writeFileSync(join(fullPath, file), multipleCards.integration);
+            writeFileSync(join(fullPath, 'adaptive-' + file), multipleCards.adaptive);
+
+            const manifestPath = './webapp/manifest.json';
+            const oManifest = JSON.parse(await manifest.getString());
+
+            if (!oManifest['sap.cards.ap']) {
+                oManifest['sap.cards.ap'] = {};
+            }
+            const integrationCard = manifests.find((card: any) => card.type === 'integration');
+            const entitySet = integrationCard.entitySet;
+            oManifest['sap.cards.ap'].embeds ??= {};
+            oManifest['sap.cards.ap'].embeds[floorplan] ??= {
+                default: entitySet,
+                manifests: {}
+            };
+            oManifest['sap.cards.ap'].embeds[floorplan]['default'] ??= entitySet;
+            oManifest['sap.cards.ap'].embeds[floorplan]['manifests'] ??= {};
+            oManifest['sap.cards.ap'].embeds[floorplan]['manifests'][entitySet] ??= [
+                {
+                    'localUri': fullPath
+                }
+            ];
+            writeFileSync(manifestPath, JSON.stringify(oManifest, null, 2));
+            res.status(201).send(`Files were updated/created`);
+        } catch (err) {
+            res.status(500).send(`Files could not be created/updated.`);
+        }
+    });
+
+    /**
+     * Route to store i18n properties
+     * All the new properties are added at the end of the i18n file
+     */
+    router.post(ApiRoutes.i18nStore, async (req: Request, res: Response) => {
+        try {
+            const oManifest = JSON.parse(await manifest.getString());
+            const i18nFilePath = oManifest['sap.app'].i18n || '/i18n/i18n.properties';
```

Comment:
Module `@sap-ux/i18n` can be used to modify i18n files. Just a suggested, we can consider this later.